core/page-assets/data-managers/base.php

```diff
@@ -0,0 +1,314 @@
+	 */
+	protected $content_type;
+
+	/**
+	 * @var string
+	 */
+	protected $assets_category;
+
+	/**
+	 * @var array
+	 */
+	private $assets_config;
+
+	/**
+	 * @var array
+	 */
+	private $files_data;
+
+	/**
+	 * Get Asset Content.
+	 *
+	 * Responsible for extracting the asset data from a certain file.
+	 * Will be triggered automatically when the asset data does not exist, or when the asset version was changed.
+	 *
+	 * @since 3.3.0
+	 * @access public
+	 *
+	 * @return string
+	 */
+	abstract protected function get_asset_content();
+
+	/**
+	 * Get Asset Key.
+	 *
+	 * The asset data will be saved in the DB under this key.
+	 *
+	 * @since 3.3.0
+	 * @access protected
+	 *
+	 * @return string
+	 */
+	protected function get_key() {
+		return $this->assets_config['key'];
+	}
+
+	/**
+	 * Get Relative Version.
+	 *
+	 * The asset data will be re-evaluated according the version number.
+	 *
+	 * @since 3.3.0
+	 * @access protected
+	 *
+	 * @return string
+	 */
+	protected function get_version() {
+		return $this->assets_config['version'];
+	}
+
+	/**
+	 * Get Asset Path.
+	 *
+	 * The asset data will be extracted from the file path.
+	 *
+	 * @since 3.3.0
+	 * @access protected
+	 *
+	 * @return string
+	 */
+	protected function get_file_path() {
+		return $this->assets_config['file_path'];
+	}
+
+	/**
+	 * Get Config Data.
+	 *
+	 * Holds a unique data relevant for the specific assets category type.
+	 *
+	 * @since 3.3.0
+	 * @access protected
+	 *
+	 * @return string|array
+	 */
+	protected function get_config_data( $key = '' ) {
+		if ( isset( $this->assets_config['data'] ) ) {
+			if ( $key ) {
+				if ( isset( $this->assets_config['data'][ $key ] ) ) {
+					return $this->assets_config['data'][ $key ];
+				}
+
+				return '';
+			}
+
+			return $this->assets_config['data'];
+		}
+
+		return [];
+	}
+
+	/**
+	 * Set Asset Data.
+	 *
+	 * Responsible for setting the current asset data.
+	 *
+	 * @since 3.3.0
+	 * @access protected
+	 *
+	 * @return void
+	 */
+	protected function set_asset_data( $asset_key ) {
+		if ( ! isset( $this->assets_data[ $asset_key ] ) ) {
+			$this->assets_data[ $asset_key ] = [];
+		}
+
+		$this->assets_data[ $asset_key ]['content'] = $this->get_asset_content();
+		$this->assets_data[ $asset_key ]['version'] = $this->get_version();
+
+		$this->save_asset_data( $asset_key );
+	}
+
+	/**
+	 * Save Asset Data.
+	 *
+	 * Responsible for saving the asset data in the DB.
+	 *
+	 * @since 3.3.0
+	 * @access protected
+	 *
+	 * @param string $asset_key
+	 *
+	 * @return void
+	 */
+	protected function save_asset_data( $asset_key ) {
+		$assets_data = get_option( self::ASSETS_DATA_KEY, [] );
+		$content_type = $this->content_type;
+		$assets_category = $this->assets_category;
+
+		if ( ! isset( $assets_data[ $content_type ] ) ) {
+			$assets_data[ $content_type ] = [];
+		}
+
+		if ( ! isset( $assets_data[ $content_type ][ $assets_category ] ) ) {
+			$assets_data[ $content_type ][ $assets_category ] = [];
+		}
+
```

Comment:
Do this logic inside `get_saved_assets_data()`

Reply:
Done.